src/ILCompiler.Compiler/src/Compiler/VirtualMethodCallHelper.cs

```diff
@@ -56,7 +56,7 @@ private static int GetNumberOfBaseSlots(NodeFactory factory, TypeDesc owningType
 
                 // For types that have a generic dictionary, the introduced virtual method slots are
                 // prefixed with a pointer to the generic dictionary.
-                if (baseType.HasGenericDictionarySlot())
+                if (baseType.HasGenericDictionarySlot() && !doNotCountDictionarySlots)
```

Comment:
Can you make this `countDictionarySlots`? Double negation is kind of hard to parse for humans.